src/main/java/uk/gov/hmcts/dts/fact/model/Court.java

```diff
@@ -83,45 +86,85 @@ public Court(uk.gov.hmcts.dts.fact.entity.Court courtEntity) {
         this.crownLocationCode = courtEntity.getNumber();
         this.countyLocationCode = courtEntity.getCciCode();
         this.magistratesLocationCode = courtEntity.getMagistrateCode();
-        this.areasOfLaw = courtEntity
-            .getAreasOfLaw()
+        this.areasOfLaw = getAreasOfLaw(courtEntity);
+        final List<uk.gov.hmcts.dts.fact.entity.Contact> contacts = getContactEntities(courtEntity);
+        this.contacts = getContacts(contacts);
+        this.courtTypes = courtEntity.getCourtTypes().stream().map(CourtType::getName).collect(toList());
+        this.emails = getEmails(courtEntity);
+        this.openingTimes = getOpeningTimes(courtEntity);
+        this.applicationUpdates = getApplicationUpdates(courtEntity);
+        this.facilities = getFacilities(courtEntity);
+        this.addresses = courtEntity.getAddresses().stream().map(CourtAddress::new).collect(toList());
+        this.gbs = courtEntity.getGbs();
+        this.dxNumbers = getDxNumbers(contacts);
+        this.serviceAreas = courtEntity.getServiceAreas() == null ? null : getServiceAreas(courtEntity);
```

Comment:
Is returning an empty list better than a null? The `== null ? null` bit in particular made me wonder. Returning nulls all around the place: there are quite a few articles out there for why it might be good to avoid that as much as we can

Reply:
I know saw that...there is any existing stuff so I just leave it there in case the front end specially check for null